### PR TITLE
MKM: Kaya, Spirits' Justice

### PR DESCRIPTION
**Pull request by MikeS-NZ**
MKM: Add script for new MKM planeswalker Kaya, Spirits' Justice.

- Kaya, Spirits' Justice

Planeswalkers are getting more and more complicated...

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Name:Kaya, Spirits' Justice
+ManaCost:2 W B
+Types:Legendary Planeswalker Kaya
+Loyalty:3
+T:Mode$ ChangesZoneAll | ValidCards$ Creature.YouCtrl | Origin$ Battlefield,Graveyard | Destination$ Exile | TriggerZones$ Battlefield | RememberChanged$ True | Execute$ TrigCopy | StackDescription$ SpellDescription | SpellDescription$ Whenever one or more creatures you control and/or creature cards in your graveyard are put into exile, you may choose a creature card from among them. Until end of turn, target token you control becomes a copy of it, except it has flying.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt:5]: like I said RememberChanged is useless

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt:5]: same for the Descriptions

File: forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+SVar:TrigCopy:DB$ Clone | Choices$ Card.TriggeredCards+Creature | ChoiceZone$ Exile | ChoiceTitle$ Choose a creature card among the exiled cards to copy | ValidTgts$ Permanent.token+YouCtrl | TgtPrompt$ Choose target token you control | AddKeywords$ Flying | Duration$ UntilEndOfTurn | Hidden$ True | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt:6]: you're trying to call a SubAbility that doesn't exist

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt:6]: Hidden$ is not a param in the context of Clone Effect

File: forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+A:AB$ Surveil | Cost$ AddCounter<2/LOYALTY> | Amount$ 2 | Planeswalker$ True | SubAbility$ DBExile | SpellDescription$ Surveil 2, then exile a card from a graveyard.
+SVar:DBExile:DB$ ChangeZone | Origin$ Graveyard | Destination$ Exile | ChangeType$ Card | ChangeNum$ 1 | SelectPrompt$ Select a card from a graveyard | Mandatory$ True | Hidden$ True
+A:AB$ Token | Cost$ AddCounter<1/LOYALTY> | Planeswalker$ True | TokenScript$ wb_1_1_spirit_flying | SpellDescription$ Create a 1/1 white and black Spirit creature token with flying.
+A:AB$ ChangeZone | Cost$ SubCounter<2/LOYALTY> | Planeswalker$ True | ValidTgts$ Creature.YouCtrl | Origin$ Battlefield | Destination$ Exile | TgtPrompt$ Select target creature you control | SubAbility$ DBForEach | SpellDescription$ Exile target creature you control. For each other player, exile up to one target creature that player controls.
+SVar:DBForEach:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | ValidTgts$ Creature.OppCtrl | TgtPrompt$ Select up to one target creature each other player controls | TargetMin$ 0 | TargetMax$ OneEach | TargetsWithDifferentControllers$ True
```
[the remark this answers is not in the document]
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt]: Did my best to address your suggestions above, thanks for the feedback!

[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt]: you also need to adjust your OneEach value accordingly...

[the remark this answers is not in the document]
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt]: Changed the value to `PlayerCountOther`

File: forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt (hunk 1 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/kaya_spirits_justice.txt]: Fixed!

**Review by tool4ever — APPROVED** [review]

**Pull request merged**